### DISCUSSION TITLE
How to change the meta dict keys in the result file?

### DISCUSSION THREAD

**Discussion by Yifang-Yuan** [miniscope/minian#144] [Q&A] [answered]
Sorry, it might be a stupid question cz I'm really not familiar with the data structure.
If I want to change the session name or animal name after processing the data, how can I change and save it with the original format?

I can assign new session name by minian_dt["session"]="foo", but when I saves it by minian_dt.to_zarr, the folder structure seems different from what I get from save_minian in the pipeline, and I cannot open it by open_minian. 

I also tried save_minian(minian_dt, dpath), but not working, maybe I was doing it not properly, like this, I got the following error:

minian_ds=save_minian(minian_ds.rename("ds"), dpath, meta_dict=dict(session="s9170_210612_A", animal="Data"), overwrite=True)

ValueError: the first argument to .rename must be a dictionary

**Comment by phildong** [comment, marked as answer]
Hey sorry for the belated reply!

1. there's no built-in function to change the metadata of existing data on disk, since it's a bit to tricky to generalize all the re-writing cases and we'd like to avoid any data loss. The built-in way to correct for this is actually to correct it on-the-fly everytime during load time -- through the `post_process` argument to [open_minian](https://minian.readthedocs.io/en/stable/api/minian.utilities.html#minian-utilities-open_minian). So you would basically do something like this:
```
def correct_meta(ds, dpath):
    ds['session'] = 'foo'
    return ds

open_minian(..., post_process=correct_meta)
```

2. if you do want to correct the data on-disk so you don't have to worry about fixing it everytime, you can easily do so with `to_zarr` as you noted. Minian dataset is nothing else than a folder containing a bunch of zarr array, and once you put your zarr arrays in a folder you should be able to use `open_minian` on that folder. btw, you are also not limited to zarr storage backend once you're done with minian, check [this](https://minian.readthedocs.io/en/stable/tips/variables.html#saving-to-other-formats) out.

3. I'm not sure exactly what happened for your `save_minian` call, but usually that error happens when the argument (`minian_ds` in your case) is a [xr.Dataset](http://xarray.pydata.org/en/stable/generated/xarray.Dataset.html), however `save_minian` expect a [xr.DataArray](http://xarray.pydata.org/en/stable/generated/xarray.DataArray.html)
